contributors/design-proposals/propagation.md

```diff
@@ -142,13 +164,38 @@ distros.
 1. (From @euank) Changing those mountflags may make docker even less stable,
 this may lock up kernel accidentally or potentially leak mounts.
 
+1. (From @jsafrane) Typical container that needs to mount something needs to
+see host's `/dev` and `/sys` as HostPath volumes. This would make them shared
+without any way to opt-out. Docker creates a new `/dev/shm` in the
+container, which gets propagated to the host, shadowing host's `/dev/shm`.
+Similarly, systemd running in a container is very picky about `/sys/fs/cgroup`
+and something prevents it from starting if `/sys` is shared.
 
 ## Decision
 
-We will take 'Make HostPath shared for privileged containers, slave for
-non-privileged', an environment check and an WARNING log will be emitted about
-whether propagation mode is supported.
-
+* We will take 'Add an option in HostPathVolumeSource API'
+  * With an alpha feature gate in 1.8.
+  * Only privileged containers can use `rshared` mount propagation.
+    * When non-privileged container uses `rshared` HostPath, it silently
```

Comment:
Consider a pod that has two containers, one privileged "sidecar" and one unprivileged mysql. The sidecar one just mounts some volume for the mysql and unmounts it when the pod dies. If I add admission handler to kubelet, it will see a pod with an unprivileged container using a HostPath ( EmptyDir would be better in this case) with rshared propagation and it will reject it.